### DISCUSSION TITLE
Please specify a platform for this target in your Podfile... Podfile does not exist!

### DISCUSSION THREAD

**Discussion by raenk** [codemagic-ci-cd/codemagic-docs#1917] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I develop in linux using VSCode, building and deploying to the AppStore via codemagic.

I have a published App, both stores (apple/android) and just recently updated its dependencies and add this couple webview_flutter and package_info_plus. Nothing complex.

PlayStore: Built locally without trouble for android and the updated build was published.

AppStore: Failed. The error:

```Automatically assigning platform `iOS` with version `11.0` on target `Runner` because no platform was specified. Please specify a platform for this target in your Podfile.```

However, the Podfile does not exist and flutter does not seem to consider it necessary somehow as it won't create it.

### Expected results

I expect to have a successful build and deployment as with the previous build.

### Actual results

Not building asking something that can't be done or at least not straight forward.

### Build id (optional)

64c15b234e70f964891aaa69

**Comment by kulakarla** [comment]
Hi @raenk 

Inside your repository, run ``flutter create .``
This command will create all the missing files, including the Podfile with various flutter specific pod scripts. Commit these files to your repository. Then, inside the Podfile you have this line: ``# platform :ios, '11.0'
``. Uncomment it, and change the version according to your dependencies needs.

This should fix the problem. If that is not the case however, let me know.

**Reply by raenk** [reply]
Hi @kulakarla 

Thanks. Forgot to mention that I already tried ```flutter create --platforms ios``` and it does not get created

**Reply by kulakarla** [reply]
Have you tried creating the Podfile manually and adding ``platform: ios, '12.0'``line there?

**Reply by raenk** [reply]
No, not really. I have no idea what information the file should have or if there's any other stuff needed like directories or additional files for my specific project.

**Reply by kulakarla** [reply, marked as answer]
@raenk 

Alright, took a second look at the errors and seems like the main culprit here is actually this error: 

```
Unable to find a target named RunnerTests in project Runner.xcodeproj, did find Runner
```

The error is caused by the fact that you are building using newer versions of Flutter, but the project itself was created using an older version. Flutter has changed their `ios` folder setup, which is why the older projects run into this error with ios when using newer versions of Flutter.

There are three methods in how to fix it:

1. On your local machine, using the latest stable flutter for example, delete `ios`folder, then run `flutter create .`. This does not create a `Podfile` on Windows/Linux machines, however, it changes the other `.xcodeproj` etc. files. Commit this refreshened `ios` folder to your repository and run a build.

2. The simplest - downgrade your Flutter verison. E.g 3.7.2 or whatever you were using before. However, I don't know all about your dependencies, so its up to you if you can do that.

3. This is a multi step process and needs you to enter our VM-s, whether by SSH or VNC connection.
* Start a build, enable SSH connection
* SSH or VNC into the machine
* When in the machine, navigate to `/Users/builder/clone/ios`
* There, a `Podfile` should be generated. Copy the contents of this `Podfile`, and create a `Podfile` locally inside your `ios`directory.
* Paste the contents of the VM-s `Podfile` to your locally created one. 
* Now, inside the `Podfile`, find this part and comment out these lines which I've marked:
```
target 'Runner' do
 use_frameworks!
 use_modular_headers!

 flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
 # target 'RunnerTests' do <-- MAKE SURE THIS LINE IS COMMENTED OUT
  # inherit! :search_paths <-- MAKE SURE THIS LINE IS COMMENTED OUT
 # end <-- MAKE SURE THIS LINE IS COMMENTED OUT
end
```
* Commit changes
* Run a build


One of these solutions should do the job for you. Let me know if it doesn't fix the problem.

**Reply by raenk** [reply]
@kulakarla many many thanks!

The option 3 did the trick. I tried #1 first, but it complained about certificates at the signing stage. I assumed that, wiping the iOS directory took away more than needed, so I didn't want to deal with that and throw additional variables to the already problematic equation. So, aborted this option and tried #3.

Worked perfectly!

Last question: Do I need to keep this Podfile as is for later releases or, since the built was successful, the next releases won't need this anymore?

**Reply by kulakarla** [reply]
@raenk 

Keep the Podfile for future releases. Reason being, the error stems from the fact that with no existing Podfile, the new Flutter version will still always generate a "faulty" Podfile which mismatches your "_outdated_" ios folder, ultimately causing your build to fail.

I'm glad your finally found your answer, and contact us again if you were to run into any problems. I will also link this thread as a reply to your StackOverflow thread.
